Implement #80

phlprcks opened the pull request:
as discussed some tests fail as they were previously not tested on the dev branch where I pulled from...
All merge conflicts should have been resolved.

Commit:
added bugs and intention revealing linting tests

Commit:
fixed some linting errors and deleted unneeded linting rules

Commit:
added tests from misc category

Commit:
added tests from possible bug category

Commit:
fixed linting errors

Commit:
Merge development into implement-#80

Commit:
some more test running

Commit:
Merge branch 'development' into implement-#80

Commit:
fixed some tests

sinnaj-r commented on packages/AlgernonTests.package/ALGTest.class/instance/nameOfFirstResult.st:
```diff
@@ -1,3 +0,0 @@
-helpers
-nameOfFirstResult
```
The Removal of this file/method creates the errors in the tests. Was there a reason you removed it ?

phlprcks replied:
no - must have been some merge errors as I initially branched from a old dev branch...

sinnaj-r commented on packages/AlgernonTests.package/ALGTestUI.class/instance/testBreadcrumbDoesHanldeMouseEvents.st:
```diff
@@ -0,0 +1,5 @@
+testing
+testBreadcrumbDoesHanldeMouseEvents
+	"for ticket 11"
+	self assert: (self algerInstance breadcrumb handlesMouseDown: nil).
```
`algerInstance` is now `algInstance`. (Therefore this test is failing)

phlprcks replied:
That test was apearently deleted and persisted in my branch. We now have the ...doesNotHandle...

Commit:
fixed requested changes

Commit:
removed wrong test

Commit:
removes depreciated methods

phlprcks committed:
revokes unwanted changes to readme

readme on dev was changed meanwhile and we don't want to overriede those changes.

sinnaj-r commented on packages/Algernon.package/ALGResultItemMorph.class/instance/icon.st:
```diff
@@ -1,3 +1,3 @@
-accessing
+thumbnail
```
See "icon"-comment.

sinnaj-r commented on packages/Algernon.package/ALGStandardToolSetMorphType.class/class/additionalToolEntries.st:
```diff
@@ -1,4 +1,4 @@
-algertem lists
+as yet unclassified
```
Unclassified :D

sinnaj-r commented on packages/Algernon.package/ALGAction.class/instance/name.st:
```diff
@@ -1,4 +1,4 @@
-accessing
+testing
```
See "icon"-comment.

sinnaj-r commented on packages/Algernon.package/ALGResultItemMorph.class/instance/icon..st:
```diff
@@ -1,3 +1,3 @@
-accessing
+thumbnail
```
I think accessors should be still be in their own category.

phlprcks replied:
just fyi - some of those classifications were made bc there is a linting test that checks if the categories match the category of the methods in the super classes. Appereantly there is somewhere very up in the hirarchy some weird categorization so that I decided to not run this lint in the end. But yes, completely right I missed to revoke some of those changes - so thanks 👍

sinnaj-r commented on packages/Algernon.package/ALGStandardToolSetMorphType.class/class/algertemList.st:
```diff
@@ -1,4 +1,4 @@
-algertem lists
+as yet unclassified
```
unclassified

sinnaj-r commented on packages/Algernon.package/ALGStandardToolSetMorphType.class/instance/typeName.st:
```diff
@@ -1,3 +1,3 @@
-accessing
+actions
```
Has nothing to do with actions

sinnaj-r commented on packages/Algernon.package/ALGStandardToolSetMorphType.class/instance/name.st:
```diff
@@ -1,3 +1,3 @@
-accessing
+actions
```
Has nothing to do with actions

sinnaj-r commented on packages/Algernon.package/ALGWorldMenuMorphType.class/instance/createListOfAlgeractions.st:
```diff
@@ -1,4 +1,4 @@
-accessing
+accessors
```
`accessing` would be the correct squeak term, but I don't think this is an accessors one way or the other.

sinnaj-r commented on packages/Algernon.package/ALGWorldMenuMorphType.class/instance/createMorph..st:
```diff
@@ -1,4 +1,4 @@
-actions
+accessors
```
See createListOfAlgeractions.st comment

sinnaj-r commented on packages/AlgernonTests.package/ALGFakeKeyboardEvent.class/instance/initialize.st:
```diff
@@ -1,4 +1,4 @@
-accessing
+initialize
```
`initialize-release`

sinnaj-r commented on packages/AlgernonTests.package/ALGFakeKeyboardEvent.class/instance/isKeyUp.st:
```diff
@@ -1,3 +1,3 @@
-accessing
+testing
```
Everything in this FakeKeyboard is "testing" - therefore `accessing` is okay.

sinnaj-r commented on packages/AlgernonTests.package/ALGFakeKeyboardEvent.class/instance/keyString.st:
```diff
@@ -1,3 +1,3 @@
-accessing
+keyboard
```
See isKeyUp.st comment

sinnaj-r commented on packages/AlgernonTests.package/ALGTestAction.class/instance/testPrintIt.st:
```diff
@@ -0,0 +1,18 @@
+testing
+testPrintIt
```
Same as `testDoIt.st`.

BenBals replied:
Same comment from me too.

sinnaj-r commented on packages/AlgernonTests.package/ALGTestAction.class/instance/testDoIt.st:
```diff
@@ -0,0 +1,7 @@
+testing
+testDoIt
```
I don't know if this is your test, but nethertheless, this should be moved to on of the  ALGTest Subclasses.
(Because we don't want to test search behaviour and initialise ALGCores outside of ALGTest, as the cleanup procedures etc. are located there)

Additionally I have the feeling, that we already have a test for doit 🤔.

BenBals replied:
I think the test got deleted/moved, but this PR will add it again, because its starting point is so old. Please just remove it. It seems like a merge went wrong or something.

sinnaj-r reviewed (changes requested):
First of all: Good job! I can imagine that implementing/fixing the lints was not that exiting. 😃 
Since you have done quite a lot of new categorizations, I paid special attention to them during the review, and noticed some small things - so don't be shocked by the number of comments.

BenBals commented on packages/Algernon.package/ALGAction.class/class/validVerbs.st:
```diff
@@ -1,4 +1,4 @@
-constants
+as yet unclassified
```
See Jannis' commens.

BenBals commented on packages/Algernon.package/ALGWorldMenuMorphType.class/README.md:
```diff
@@ -1 +1 @@
-An ALGWorldMenuMorphType represents a Morph that appears in the world menu 'open' menu. The typeObject for this ALGType should be in the same format as that in TheWorldMenu registry.
\ No newline at end of file
+An ALGTestType represents a test in the system.
```
Why did you replace this comment? That's the wrong one.

phlprcks replied:
was an unwanted change, probably in the mergin process, is now changed back

BenBals commented on packages/AlgernonTests.package/ALGLinting.class/instance/testNoAbstractClass.st:
```diff
@@ -1,3 +1,3 @@
-code-style
+possible bugs
```
Nice that you classified the lints. Good job. :+1:

BenBals commented on packages/AlgernonTests.package/ALGTestAction.class/instance/core..st:
```diff
@@ -0,0 +1,3 @@
+accessing
+core: anObject
+	core := anObject
```
These can also be deleted see `testDoIt`

phlprcks replied:
not really, this one is called by other methods

BenBals commented on packages/AlgernonTests.package/ALGTestAction.class/instance/core.st:
```diff
@@ -0,0 +1,3 @@
+accessing
+core
+	^ core
```
These can also be deleted see `testDoIt`

BenBals reviewed (changes requested):
Seems very good overall.

A merge seems to have gone wrong that re-introduces code that was supposed to be deleted.

Otherwise just very minor things around classification.

Commit:
recategorizing

phlprcks committed:
increase timeout in CI

Commit:
adding default time outs

Commit:
Merge branch 'implement-#80'

BenBals reviewed (approved):
Looks good now.
